packages/headless/src/features/result/result-analytics-actions.ts

```diff
@@ -22,4 +18,11 @@ export const logDocumentOpen = (result: Result) =>
         documentIdentifier(result)
       );
     }
-  )();
+  );
+
+/**
+ * Logs a click event with an `actionCause` value of `documentOpen`.
+ * @param result (Result) The result that was opened.
+ */
+export const logDocumentOpen = (result: Result) =>
+  logDocumentOpenThunk(result)();
```

Comment:
Why create a separate thunk function instead of invoking as was the case on line 25 of the red diff?

Reply:
For unit testing. Without doing this, we can't retrieve the `pending` action. An alternative would be to explicitly check for the action type instead.

I figured this approach wasn't too much of a problem since `logDocumentOpenThunk` isn't exported outside of `headless`.